### PR TITLE
New style travis tests

### PR DESCRIPTION
**Pull request by chillu**
- Removed PHP 5.5. build (no longer supported)
- Removed PG and SQLite tests (tested through installer nightly)
- Added code coverage and linting tests
- Fixed linting errors

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,30 +1,34 @@
...
-  - git clone git://github.com/silverstripe-labs/silverstripe-travis-support.git ~/travis-support
-  - php ~/travis-support/travis_setup.php --source `pwd` --target ~/builds/ss
-  - cd ~/builds/ss
-  - composer install
+  - "if [ \"$PHPCS_TEST\" = \"1\" ]; then pyrus install pear/PHP_CodeSniffer; fi"
+  - phpenv rehash
+  - phpenv config-rm xdebug.ini
+  - composer validate
+  - composer install --prefer-dist
+  - composer require silverstripe/admin:4.0.x-dev silverstripe/framework:4.0.x-dev silverstripe/versioned:1.0.x-dev silverstripe/config:1.0.x-dev --prefer-dist
```
Comment (sminnee) [inline .travis.yml]: * admin should be 1.0.x-dev
 * use `composer require --update-with-dependencies` to be more robust around upgrading/downgrading already installed packages

**Comment by sminnee** [comment]
Coverage is segfaulting :-(

**Comment by chillu** [comment]
Also needs to be rebased after https://github.com/silverstripe/silverstripe-reports/pull/76

**Comment by chillu** [comment]
@sminnee It's green now

File: .travis.yml [hunk 2]
```diff
@@ -1,30 +1,40 @@
-# See https://github.com/silverstripe-labs/silverstripe-travis-support for setup details
+language: php
 
 sudo: false
 
-language: php
+env:
+  global:
+    - COMPOSER_ROOT_VERSION=1.0.x-dev
```
Comment (tractorcow) [inline .travis.yml]: reports version is 4.x not 1.x.

**Comment by tractorcow** [comment]
So these new dev-only dependencies broke beta1.

**Comment by sminnee** [comment]
Sad trombone...

**Pull request merged**